lib/student.rb

```diff
@@ -0,0 +1,17 @@
+class Student
+    attr_reader :name, :age, :scores
+    def initialize(student_data)
+        @name = student_data[:name]
+        @age = student_data[:age]
+        @scores = student_data.fetch(:scores, [])
+    end
+
+    def log_score(score)
+        @scores << score
+    end
+
+    def grade
+        return 0 if @scores.empty?
+        @scores.sum.to_f / @scores.size
```

Comment:
Also it is an absolute crime that there is no average method in Ruby

Reply:
🚓

Reply:
![image](https://github.com/user-attachments/assets/3bf47fd2-9a33-4742-8b21-8c6f96a9d339)

Reply:
![](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExZzBkYmg2OWFsMXBnMDg4aTNndDlybWdjMnR4dmI3bjRsNGwxeTA4NSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/atd2NiCLAG1Xy/giphy.gif)